### PR TITLE
#14862: fp32 support in unary

### PR DESCRIPTION
**Pull request by KalaivaniMCW**
### Ticket
Link to Github Issue #14862

### Problem description
Provide context for the problem.

### What's changed
Enabled `preserve_fp32_precision` flag  for float32 input 

### Checklist
- [x] Post commit CI passes https://github.com/tenstorrent/tt-metal/actions/runs/11752673780
https://github.com/tenstorrent/tt-metal/actions/runs/11797734586
- [ ] Nightly fd https://github.com/tenstorrent/tt-metal/actions/runs/11797739127
- [ ] Model perf
- [ ] Device perf
- [ ] Demo tests
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/ttnn/unit_tests/operations/eltwise/test_fp32.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    x_torch = torch.tensor([[0.00001]], dtype=torch.float32)
+    y_torch = -x_torch
+    print("y_torch", y_torch)
+
+    x_tt = ttnn.from_torch(x_torch, dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)
+    y_tt = ttnn.neg(x_tt)
+    print("y_tt", y_tt)
+
+    print(x_torch, ttnn.to_torch(x_tt))
+    print(y_torch, ttnn.to_torch(y_tt))
```
Comment (eyonland) [inline tests/ttnn/unit_tests/operations/eltwise/test_fp32.py:24]: @KalaivaniMCW , there are no asserts in this test.   Take a look at my comment on https://github.com/tenstorrent/tt-metal/pull/14874

Comment (KalaivaniMCW) [reply tests/ttnn/unit_tests/operations/eltwise/test_fp32.py:19]: Yes it is passing with assert. @eyonland 

Screenshot 2024-11-08 at 10 29 21 PM

**Review by eyonland — APPROVED** [review]

File: tests/ttnn/unit_tests/operations/eltwise/test_fp32.py (hunk 1 above)
Comment (eyonland) [reply tests/ttnn/unit_tests/operations/eltwise/test_fp32.py:19]: Please add these asserts into your tests.

**Review by eyonland — APPROVED** [review]
Let's make sure that we don't loose any performance.  But looks good.

**Review by dmakoviichuk-tt — APPROVED** [review]
Thanks!
Please add a few more random tests for a couple of ops.

**Review by rfurko-tt — APPROVED** [review]
Thanks for enabling FP32 support for unary ops. Looks good overall.
Please create separate test for each unary op to make sure that this fix works.

**Pull request merged**